virtcontainers/clh.go

```diff
@@ -391,8 +392,42 @@ func (clh *cloudHypervisor) resizeMemory(reqMemMB uint32, memoryBlockSizeMB uint
 }
 
 func (clh *cloudHypervisor) resizeVCPUs(reqVCPUs uint32) (currentVCPUs uint32, newVCPUs uint32, err error) {
-	clh.Logger().WithField("function", "resizeVCPUs").Warn("not supported")
-	return 0, 0, nil
+	cl := clh.client()
+
+	// Retrieve the number of current vCPUs via HTTP API
+	ctx, cancel := context.WithTimeout(context.Background(), clhAPITimeout*time.Second)
+	info, _, err := cl.VmInfoGet(ctx)
+	if err != nil {
+		clh.Logger().WithField("function", "resizeVCPUs").WithError(openAPIClientError(err)).Info("[clh] VmInfoGet failed")
+		return 0, 0, openAPIClientError(err)
+	}
+	cancel()
+
+	currentVCPUs = uint32(info.Config.Cpus.BootVcpus)
+	newVCPUs = currentVCPUs
+
+	// Sanity check
+	if reqVCPUs == 0 {
+		clh.Logger().WithField("function", "resizeVCPUs").Debugf("Cannot resize vCPU to 0")
+		return currentVCPUs, newVCPUs, fmt.Errorf("Cannot resize vCPU to 0")
+	}
+	if reqVCPUs > clh.config.DefaultMaxVCPUs || reqVCPUs > uint32(info.Config.Cpus.MaxVcpus) {
+		clh.Logger().WithField("function", "resizeVCPUs").Debugf("Cannot resize vCPU to %d: exceeding the maximum amount of vCPUs (%d or %d)",
```

Comment:
please use `WithFields`, not "%d"

Reply:
@devimc Thanks for the comments. Just adjusted as you advised. One question I have was: what is the right way to break the long line into multiple lines in golang? The line I changed now is even longer, and is ready hard to read. :<

Reply:
this is a good example:

https://github.com/kata-containers/runtime/blob/fde6447c16721dc5fa3139872bc791ec08afea4e/virtcontainers/device/manager/utils.go#L67-L72

Reply:
Much easier to read now. Thanks a lot, @devimc.